test/e2e/os/arch.go

```diff
@@ -94,8 +105,16 @@ func getAmiIDFromSSM(ctx context.Context, client *ssm.Client, amiName string) (*
 }
 
 // an unknown size and arch combination is a coding error, so we panic
-func getInstanceTypeFromRegionAndArch(_ string, arch architecture, instanceSize e2e.InstanceSize) string {
-	instanceType, ok := instanceSizeToType[arch][instanceSize]
+func getInstanceTypeFromRegionAndArch(_ string, arch architecture, instanceSize e2e.InstanceSize, gpuInstance bool) string {
```

Comment:
Could we avoid using boolean arguments? It can make code less readable and less extensible. I would prefer an enum approach. Maybe something like this:
```
const (
    StandardInstance InstanceCategory = iota
    GPUInstance
)

func getInstanceTypeFromRegionAndArch(_ string, arch architecture, instanceSize e2e.InstanceSize, category InstanceCategory)
```